### PR TITLE
🔀🔖 4.5.8

### PR DESCRIPTION
**Pull request by jxn-30**
* 🐛 [core] fix loading issues
* 🐛 [eb] fix expansions countdown
* 🐛 [redesign/conversations] fix alliance messages
* ✨👮🇬🇧 UK CP 🇬🇧👮✨
* ✨ [sap] creating large scale alliance missions

### CODE REVIEW

File: src/modules/shareAlliancePost/assets/util.ts [hunk 1]
```diff
@@ -154,14 +154,21 @@ export function createIcon(
     return iconElement;
 }
 
-export function createEditBtn(editBtnClass: string) {
+export function createEditBtn(
+    editBtnClass: string[] | string = [],
```
Comment (Cr4zyc4k3) [inline src/modules/shareAlliancePost/assets/util.ts:158]: Array string or string as array. Is that necessary?

File: src/modules/shareAlliancePost/assets/util.ts [hunk 2]
```diff
@@ -154,14 +154,21 @@ export function createIcon(
...
+) {
     const editBtn = document.createElement('button');
-    editBtn.classList.add('btn', 'btn-xs', 'btn-default', editBtnClass);
-    editBtn.style.setProperty('position', 'absolute');
-    editBtn.style.setProperty('right', '0');
-    editBtn.style.setProperty('transform', 'translateY(-3px)');
+    editBtn.classList.add('btn', 'btn-xs', 'btn-default');
+    if (Array.isArray(editBtnClass)) editBtn.classList.add(...editBtnClass);
+    else editBtn.classList.add(editBtnClass);
+    if (transform) {
```
Comment (Cr4zyc4k3) [inline src/modules/shareAlliancePost/assets/util.ts:165]: Why check if its true every time?

File: src/modules/shareAlliancePost/assets/util.ts [hunk 3]
```diff
@@ -170,16 +177,29 @@ export function createEditBtn(editBtnClass: string) {
...
     const wrapper = document.createElement('div');
-    wrapper.classList.add('input-group', inputGroupClass);
-    wrapper.style.setProperty('position', 'absolute');
-    wrapper.style.setProperty('left', isMissionList ? '0' : '100%');
-    wrapper.style.setProperty('transform', 'translateY(-26px)');
+    wrapper.classList.add('input-group');
+    if (Array.isArray(inputGroupClass))
+        wrapper.classList.add(...inputGroupClass);
+    else wrapper.classList.add(inputGroupClass);
+    if (transform) {
```
Comment (Cr4zyc4k3) [inline src/modules/shareAlliancePost/assets/util.ts:198]: Same here

File: src/modules/shareAlliancePost/assets/util.ts (hunk 1 above)
Comment (jxn-30) [reply src/modules/shareAlliancePost/assets/util.ts:158]: It is "either a array of strings or a string, but by default (if the parameter is not given), it is an empty array)

File: src/modules/shareAlliancePost/assets/util.ts (hunk 2 above)
Comment (jxn-30) [reply src/modules/shareAlliancePost/assets/util.ts:165]: only default value is true but parameter can be changed

**Review by Cr4zyc4k3 — APPROVED** [review]

**Pull request merged**